lib/backend/k8s/k8s.go

```diff
@@ -551,20 +564,42 @@ func (c *KubeClient) getPolicy(k model.PolicyKey) (*model.KVPair, error) {
 	if k.Name == "" {
 		return nil, goerrors.New("Missing policy name")
 	}
-	namespace, policyName := c.converter.parsePolicyName(k.Name)
 
-	networkPolicy := extensions.NetworkPolicy{}
-	err := c.clientSet.Extensions().RESTClient().
-		Get().
-		Resource("networkpolicies").
-		Namespace(namespace).
-		Name(policyName).
-		Timeout(10 * time.Second).
-		Do().Into(&networkPolicy)
-	if err != nil {
-		return nil, resources.K8sErrorToCalico(err, k)
+	// Check to see if this is backed by a NetworkPolicy or a Namespace.
+	if strings.HasPrefix(k.Name, "np.projectcalico.org") {
+		// Backed by a NetworkPolicy. Parse out the namespace / name.
+		namespace, policyName, err := c.converter.parsePolicyNameNetworkPolicy(k.Name)
+		if err != nil {
+			return nil, errors.ErrorResourceDoesNotExist{Err: err, Identifier: k}
+		}
+
+		// Get the NetworkPolicy from the API and convert it.
+		networkPolicy := extensions.NetworkPolicy{}
+		err = c.clientSet.Extensions().RESTClient().
+			Get().
+			Resource("networkpolicies").
+			Namespace(namespace).
+			Name(policyName).
+			Timeout(10 * time.Second).
+			Do().Into(&networkPolicy)
+		if err != nil {
+			return nil, resources.K8sErrorToCalico(err, k)
+		}
+		return c.converter.networkPolicyToPolicy(&networkPolicy)
+	} else {
+		// This is backed by a Namespace.
```

Comment:
I think this also needs to handle the does-not-exist path: that's mainline(ish), so probably worth mentioning.